README.md

```diff
@@ -31,37 +36,255 @@ npm install @uploadcare/upload-client --save
 ```
 
+Or set callback function that will be called when file was uploaded:
+```javascript
+fileUpload.onUploaded = (uuid => console.log(`File "${uuid}" was uploaded.`))
+```
+
+Or when file is ready on CDN:
+```javascript
+fileUpload.onReady = (file => console.log(`File "${file.uuid}" is ready on CDN.`))
+```
+
+You can cancel file uploading and track this event:
+```javascript
+// Set callback
+fileUpload.onCancel = (() => console.log(`File uploading was canceled.`))
+
+// Cancel uploading
+fileUpload.cancel()
+```
+
+### Middle Level API
+
+Also, you can use wrappers around low level to call the API endpoints:
+
+```javascript
+import UploadClient from '@uploadcare/upload-client'
+
+const client = new UploadClient({publicKey: 'YOUR_PUBLIC_KEY'})
+const api = client.api
+const directUpload = api.base(fileData) // fileData must be `Blob` or `File` or `Buffer`
+
+directUpload
+  .then(data => console.log(data.file))
+
+directUpload.onProgress = (progressEvent) => console.log(progressEvent.loaded / progressEvent.total)
+
+// Also you can cancel upload:
+directUpload.cancel()
+
+// and set callback to track cancel event:
+directUpload.onCancel = () => console.log('File upload was canceled.') 
+```
+
+List of all available low level API methods:
+
+```typescript
+interface UploadAPIInterface {
+  request(options: RequestOptionsInterface): Promise<RequestResponse>
+
+  base(data: FileData, settings?: SettingsInterface): BaseThenableInterface<BaseResponse>
+
+  info(uuid: Uuid, settings?: SettingsInterface): CancelableThenableInterface<FileInfoInterface>
+
+  fromUrl(sourceUrl: Url, settings?: SettingsInterface): CancelableThenableInterface<FromUrlResponse>
+
+  fromUrlStatus(token: Token, settings?: SettingsInterface): CancelableThenableInterface<FromUrlStatusResponse>
+
+  group(files: Uuid[], settings: SettingsInterface): CancelableThenableInterface<GroupInfoInterface>
+
+  groupInfo(id: GroupId, settings: SettingsInterface): CancelableThenableInterface<GroupInfoInterface>
+
+  multipartStart(file: FileData, settings: SettingsInterface): CancelableThenableInterface<MultipartStartResponse>
+
+  multipartUpload(file: FileData, parts: MultipartPart[], settings: SettingsInterface): BaseThenableInterface<any>
+
+  multipartComplete(uuid: Uuid, settings: SettingsInterface): CancelableThenableInterface<FileInfoInterface>
+}
+```
+
+### Low Level API
+
+If you want to use our low level API, you can do that something like this:
+
+```javascript
+import UploadClient from '@uploadcare/upload-client'
+
+const client = new UploadClient({publicKey: 'YOUR_PUBLIC_KEY'})
+
+client.api.request({path: 'info', query})
+  .then(response => console.log(response.data))
+```
+
+### Settings
+
+#### publicKey: string
+
+The main use of a `publicKey` is to identify a target project for your uploads. 
+It is required when using Upload API.
+
+#### baseCDN: string
+
+Defines your schema and CDN domain. Can be changed to one of 
+the predefined values or your custom CNAME. 
```

Comment:
stating "one of the predefined values" without listing those values is unclear

Reply:
For some customers we do have custom CDN and Upload API URL bases - custom CNAME's.
This description should be changed so it will clearly describe why  you want to change this option.